modules/build/src/main/scala/scala/build/compiler/BloopCompilerMaker.scala

```diff
@@ -30,6 +33,20 @@ final class BloopCompilerMaker(
           threads,
           logger.bloopRifleLogger
         )
-    new BloopCompiler(createBuildServer, 20.seconds, strictBloopJsonCheck)
+
+    Try(new BloopCompiler(createBuildServer, 20.seconds, strictBloopJsonCheck))
+      .toEither
+      .left.flatMap {
+        case e if offline =>
+          e.getCause match
+            case _: FetchingDependenciesError =>
+              logger.debug("Couldn't fetch Bloop from cache, fallback to using scalac")
```

Comment:
I wonder if it wouldn't be better to print this as a warning, instead of a debug message... I'd want to know if `scalac` was used instead of `bloop` implicitly in my build.
Also, I'd be explicit that this change happens because `--offline` was passed.

So something along the lines of:
```suggestion
              logger.message("Warning: Bloop could not be fetched from the local cache, as offline mode is on, using scalac as fallback")
```